### PR TITLE
Test add flow/iulian

### PR DESCRIPTION
**Pull request by Iulik1989**

### CODE REVIEW

File: src/pages/About/MenuTest/index.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import { MenuItemEmailTest, MenuItemIconTest, MenuItemNameTest, MenuItemTest, MenuTest1 } from '../style'
+
+const MenuTest = ({ query, exclude, list, onSelect, top, left }) => {
+  const options = list.filter((user) => {
+    return user.email.includes(query) && !exclude.includes(user.id)
```
Comment (MihailGorceagEndava) [inline src/pages/About/MenuTest/index.jsx:6]: from BE you already receive filtered data by query, that is sent to the BE, so you just need to display the result.

File: src/pages/About/MenuTest/index.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  })
+  if (!options.length || !query) {
+    return null
+  }
+
+  return (
+    <MenuTest1 top={`${top}px`} left={`${left}px`}>
```
Comment (MihailGorceagEndava) [inline src/pages/About/MenuTest/index.jsx:13]: please rename to a better name

File: src/pages/About/style.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
+import styled from 'styled-components'
+
+export const MenuTest1 = styled.ul`
```
Comment (MihailGorceagEndava) [inline src/pages/About/style.js:3]: you don't really need ul, you can use just div

File: src/pages/About/index.jsx [hunk 4]
```diff
@@ -1,31 +1,161 @@
...
+  useEffect(() => {
+    if (searchQuery.length >= searchStartLength) {
+      getCoProposers()
+    }
+  }, [searchQuery])
+
   useEffect(() => {
     getSite()
-  }, [])
+  }, [site])
```
Comment (MihailGorceagEndava) [inline src/pages/About/index.jsx:65]: you don't need this dependency, it should behave as componentDidMount

File: src/pages/About/index.jsx [hunk 5]
```diff
@@ -1,31 +1,161 @@
...
+        <MenuTest
+          query={searchQuery}
+          exclude={excludeIds}
+          onSelect={handleSelect}
+          top={menuPos.y}
+          left={menuPos.x}
+          list={potentialCoProposers}
+        />
+      </ChipDropdownTest>
+      <Button onClick={addNewCoProposer} isDisabled={!site.name}>
```
Comment (MihailGorceagEndava) [inline src/pages/About/index.jsx:152]: please create validation function, result of which is used to set isDisabled on this button. It should check site.name and also validity of chips

File: src/pages/About/index.jsx [hunk 6]
```diff
@@ -1,31 +1,161 @@
-import React, { useState, useEffect } from 'react'
+import React, { useState, useEffect, useRef } from 'react'
 import { useHistory, useParams } from 'react-router-dom'
 import Button from 'components/Button'
-import { requestGetSite } from 'api/site'
+import { requestAddNewCoProposer, requestGetPotentialCoProposers, requestGetSite, requestUpdateSite } from 'api/site'
+import ClearIcon from 'assets/icons/clear-icon'
+import { ChipDropdownTest, ChipTest, InputTest, ValueTest } from './style'
+import MenuTest from './MenuTest'
+import Input from '../../components/Input'
```
Comment (MihailGorceagEndava) [inline src/pages/About/index.jsx:8]: you can just write 'components/Input'

File: src/pages/About/MenuTest/index.jsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import { MenuItemEmailTest, MenuItemIconTest, MenuItemNameTest, MenuItemTest, MenuTest1 } from '../style'
+
+const MenuTest = ({ query, exclude, list, onSelect, top, left }) => {
```
Comment (MihailGorceagEndava) [inline src/pages/About/MenuTest/index.jsx:4]: this component should not care about query, exclude. it just receives list and applies onSelect

File: src/pages/About/index.jsx [hunk 8]
```diff
@@ -1,31 +1,161 @@
...
+    }
+    coProposerFetchAbortController = new AbortController()
+    requestGetPotentialCoProposers({ searchQuery, excludeIds }).then((data) => {
+      setPotentialCoProposer(data.potentialProposers)
+    })
+  }
+
+  function addNewCoProposer() {
+    const siteId = site.id
+    const newProposers = [...excludeIds]
```
Comment (MihailGorceagEndava) [inline src/pages/About/index.jsx:36]: it's definitely incorrect, exclude ids are used to filter data from requestGetPotentialCoProposers

File: src/pages/About/index.jsx [hunk 9]
```diff
@@ -1,31 +1,161 @@
...
 
   function goBack() {
     history.goBack()
   }
 
+  function handleChange(e) {
+    setSearchQuery(e.target.value)
+  }
+
+  function handleRemove(id) {
```
Comment (MihailGorceagEndava) [inline src/pages/About/index.jsx:75]: you don't need to filter anything, just do call to BE, and after success response - remove it from list on FE

File: src/pages/About/MenuTest/index.jsx (hunk 1 above)
Comment (Iulik1989) [reply src/pages/About/MenuTest/index.jsx]: i need it to remove items that were already selected and are in chips

File: src/features/AddCoProposerMenu/Menu/PotentialCoProposerMenuList.jsx [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+} from '../style'
+
+const PotentialCoProposerMenuList = ({ query, chips, list, onSelect, top, left }) => {
+  const menuRef = useRef()
+
+  const startSearchLength = 3
+  const excluded = chips.map((chip) => {
+    return chip.email
+  })
+  const options = list.filter((user) => {
```
Comment (MihailGorceagEndava) [inline src/features/AddCoProposerMenu/Menu/PotentialCoProposerMenuList.jsx:18]: please use endpoint params, to avoid filtering on FE

**Review by MihailGorceagEndava — APPROVED** [review]

**Pull request closed without merging**